src/vmm/src/persist.rs

```diff
@@ -247,6 +251,38 @@ fn snapshot_memory_to_file(
     }
 }
 
+/// Validates that snapshot CPU vendor matches the host CPU vendor.
+#[cfg(target_arch = "x86_64")]
+pub fn validate_cpu_vendor(
```

Comment:
nit: explicitly name this function `validate_x86_64_cpu_vendor()` since the code is very specific to `x86_64`

Reply:
makes sense.

Reply:
done